docs/lab/05/index.md

```diff
@@ -0,0 +1,224 @@
+slug: /lab/05
+---
+
+# 05 - Asynchronous Development
+
+This lab will teach you the principles of asynchronous programming, and its application in Embassy-rs.
+
+
+## Resources
+
+TBD
+
+## Asynchronous functions
+
+Up to now, during the labs, we've seen that, in order to be able to do multiple different actions "at once", we would use *tasks*. We would let the `main` function run, while also doing another action seemingly "in parallel" inside of another task. 
+Let's take the following example: if we want to blink an LED every second while also waiting for a button press to do something else, we would need to spawn a new task in which we would wait for the button press, while blinking the LED in the `main` function. 
+
+When thinking of how exactly this works, you would probably think that the task is running on a separate *thread* than the `main` function. Usually this would be the case when developing a normal computer application. Since only one thread can independently run per processor core (without a preemptive OS), that means that, since we are using only one core of the RP2040 (which actually has only 2), we would only be able to run **one thread at a time**. So how exactly does the task wait for the button press in parallel with the LED blinking? 
+Short answer is: it doesn't. In reality, both functions runs asynchronously. 
+
+### Tasks
+
+A task in Embassy is represented by an *asynchronous function*. Asynchronous functions are different from normal functions, in the sense that they allow asynchronous code execution. Let's take an example from the previous lab:
+```rust
+#[embassy_executor::task]
+async fn button_pressed(mut led: Output<'static, PIN_X>, mut button: Input<'static, PIN_X>) {
+    loop {
+	info!("waiting for button press");
+        button.wait_for_falling_edge().await;
+    }
+}
+
+#[embassy_executor::main]
+async fn main(spawner: Spawner) {
+    let peripherals = embassy_rp::init(Default::default());
+
+    let button = Input::new(peripherals.PIN_X, Pull::Up);
+    let led2 = Output::new(peripherals.PIN_X, Level::Low);
+
+    spawner.spawn(button_pressed(led2, button)).unwrap();
+
+    let mut led = Output::new(peripherals.PIN_X, Level::Low);
+
+    loop {
+        led.toggle();
+
+        Timer::after_millis(200).await;
+    }
+}
+```
+In this example, we notice that both the `button_pressed` and `main` functions are declared as `async`, telling the compiler to treat them as asynchronous functions. Inside the `main` function (which is also a task, actually), we blink the LED:
+
+```rust
+loop {
+    led.toggle();
+
+    Timer::after_millis(200).await;
+}
+```
+
+## `await` keyword
+
+After setting the timer, our `main` function would need to wait until the alarm fires after 200 ms. Instead of just waiting and blocking the current and *only* thread of execution, it could allow the thread to do another action in the meantime. This is where the `await` keyword comes into play.
```

Comment:
```suggestion
After setting the timer, our `main` function would need to wait until the alarm fires after 200 ms. Instead of just waiting and blocking the current and *only* thread of execution, it could allow the thread to do another action in the meantime. This is where the `.await` keyword comes into play.
```